### PR TITLE
Support for pandas TimeStamp

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
treat pandas timestamp as datetime 
and fix for DAG too often redraw

### CODE REVIEW

**Review by dinhlongviolin1 — DISMISSED** [review]
LGTMeeeeeeeeee

File: frontend/taipy/src/ScenarioDag.tsx [hunk 1]
```diff
@@ -133,14 +142,23 @@ const ScenarioDag = (props: ScenarioDagProps) => {
         if (displayModel) {
             setScenarioId(displayModel[0]);
             // populate model
-            doLayout = populateModel(displayModel, model);
+            doLayout = populateModel(addStatusToDisplayModel(displayModel, taskStatuses), model);
+        }
+        const rects = engine.getModel() && engine
```
Comment (FabienLelaquais) [inline frontend/taipy/src/ScenarioDag.tsx:147]: Oh gosh!

File: taipy/gui_core/_adapters.py [hunk 2]
```diff
@@ -182,10 +181,8 @@ def __get_data(self, dn: DataNode):
                     return (None, None, True, None)
                 val_type = (
                     "date"
-                    if "date" in type(value).__name__
+                    if "date" in type(value).__name__.lower() or "timestamp" in type(value).__name__.lower()
```
Comment (FabienLelaquais) [inline taipy/gui_core/_adapters.py:184]: 😂

File: frontend/taipy/src/ScenarioDag.tsx [hunk 3]
```diff
@@ -133,14 +142,23 @@ const ScenarioDag = (props: ScenarioDagProps) => {
...
+            .getModel()
+            .getNodes()
+            .reduce((pv, nm) => {
```
Comment (FabienLelaquais) [inline frontend/taipy/src/ScenarioDag.tsx:150]: No issue with initial value for pv?
If this works, I guess not.

File: frontend/taipy/src/ScenarioDag.tsx [hunk 4]
```diff
@@ -133,14 +142,23 @@ const ScenarioDag = (props: ScenarioDagProps) => {
...
+        if (hasPos) {
+            model.getNodes().forEach(nm => rects[nm.getID()] && nm.setPosition(rects[nm.getID()]));
         }
         engine.setModel(model);
-        // Block deletion
-        //engine.getActionEventBus().registerAction(new DeleteItemsAction({ keyCodes: [1] }));
         model.setLocked(true);
-        doLayout && setTimeout(relayout, 500);
-    }, [displayModel, engine, relayout, onClick]);
+        doLayout && !hasPos && setTimeout(relayout, 500);
```
Comment (FabienLelaquais) [inline frontend/taipy/src/ScenarioDag.tsx:160]: 500ms is usually pretty slow for a user interaction.
Can this be smaller?

**Review by FabienLelaquais — APPROVED** [review]
👓niiice

File: frontend/taipy/src/ScenarioDag.tsx (hunk 3 above)
Comment (FredLL-Avaiga) [reply frontend/taipy/src/ScenarioDag.tsx:150]: No

File: frontend/taipy/src/ScenarioDag.tsx (hunk 4 above)
Comment (FredLL-Avaiga) [reply frontend/taipy/src/ScenarioDag.tsx:160]: Not sure
I remember trying ...

**Pull request merged**